libs/vector-map/src/Data/VMap/KVVector.hs

```diff
@@ -259,20 +261,50 @@ removeDuplicates f mv
               then do
                 cur@(ck, cv) <- VGM.read mv curIx
                 if ck == pk
-                  then goMoved lastIx (ck, f ck cv pv) (curIx + 1)
+                  then goMoved lastIx (ck, f ck pv cv) (curIx + 1)
                   else goMoved (lastIx + 1) cur (curIx + 1)
               else pure $ VGM.slice 0 (lastIx + 1) mv
           goUnmoved (pk, pv) curIx
             | curIx < n = do
                 cur@(ck, cv) <- VGM.read mv curIx
                 if ck == pk
-                  then goMoved (curIx - 1) (ck, f ck cv pv) (curIx + 1)
+                  then goMoved (curIx - 1) (ck, f ck pv cv) (curIx + 1)
                   else goUnmoved cur (curIx + 1)
             | otherwise = pure mv
       x0 <- VGM.read mv 0
       goUnmoved x0 1
 {-# INLINE removeDuplicates #-}
 
+unionWithKey ::
```

Comment:
Is in the nature of a  `KVVector `  to have its elements sorted and distinct?  (which seem to be the premises of the implementation)